### PR TITLE
Remove disk interface, tests

### PR DESCRIPTION
**Pull request by slevenick**
Removes disk and region disk `interface` field as it is not useful and will be removed in the future. 

Resolves: https://github.com/hashicorp/terraform-provider-google/issues/10006
Resolves: https://github.com/hashicorp/terraform-provider-google/issues/9993

If this PR is for Terraform, I acknowledge that I have:

- [ ] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [ ] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-downstream-tools), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [ ] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/third_party/terraform/tests) (for handwritten resources or update tests).
- [ ] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/master/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [ ] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:deprecation
compute: deprecated `interface` field on `google_compute_disk` and `google_compute_region_disk` (beta only)
```

### CODE REVIEW

File: mmv1/products/compute/api.yaml [hunk 1]
```diff
@@ -3285,15 +3285,15 @@ objects:
...
+      - !ruby/object:Api::Type::String
         name: 'interface'
         min_version: 'beta'
+        # interface is removed using url_param_only to preserve schema definition
+        # and prevent sending or reading in API requests
+        url_param_only: true
+        deprecation_message: This field is no longer in use, disk interfaces will be automatically determined on attachment.
         description: |
           Specifies the disk interface to use for attaching this disk, which is either SCSI or NVME. The default is SCSI.
-        default_value: :SCSI
```
Comment (rileykarson) [inline mmv1/products/compute/api.yaml:3293]: This is unsafe to remove, right? It'll cause a diff from `"SCSI"` -> `""` for anyone using it through the default value.

File: mmv1/products/compute/api.yaml [hunk 2]
```diff
@@ -3285,15 +3285,15 @@ objects:
...
           If an unsupported value is requested, the error message will list
           the supported values for the caller's project.
-      - !ruby/object:Api::Type::Enum
+      - !ruby/object:Api::Type::String
         name: 'interface'
         min_version: 'beta'
+        # interface is removed using url_param_only to preserve schema definition
+        # and prevent sending or reading in API requests
+        url_param_only: true
+        deprecation_message: This field is no longer in use, disk interfaces will be automatically determined on attachment.
```
Comment (rileykarson) [inline mmv1/products/compute/api.yaml:3294]: What action can a user take to resolve the deprecation message?

File: mmv1/products/compute/api.yaml (hunk 1 above)
Comment (slevenick) [reply mmv1/products/compute/api.yaml:3293]: Oh, yeah that would be a disaster. Good catch!

File: mmv1/products/compute/api.yaml (hunk 2 above)
Comment (slevenick) [reply mmv1/products/compute/api.yaml]: Added note to remove it.

Added a DSF to always suppress diffs on this field, allows users to remove this if they had specified NVME without adding ignore_diffs

File: mmv1/third_party/terraform/utils/common_diff_suppress.go.erb [hunk 3]
```diff
@@ -179,3 +179,8 @@ func durationDiffSuppress(k, old, new string, d *schema.ResourceData) bool {
 	}
 	return oDuration == nDuration
 }
+
+// Suppress all diffs, used for Disk.Interface which is a nonfunctional field
+func alwaysDiffSuppress(_, _, _ string, _ *schema.ResourceData) bool {
+	return true
```
Comment (rileykarson) [inline mmv1/third_party/terraform/utils/common_diff_suppress.go.erb:185]: The behaviour of a DSF like this surprised me, so I figure I'll note it- whatever value is stored in state will always be returned. That seems fine here, although we'll need to clear out user states in `4.0.0` when we remove the field using a stateupgrader.

**Review by rileykarson — APPROVED** [review]
Cool- I believe this to be safe now. If you want to be extra thorough you could simulate the upgrade locally and make sure this works for users with various usages (specifying the default. specifying a non-default, specifying nothing)

File: mmv1/third_party/terraform/utils/common_diff_suppress.go.erb (hunk 3 above)
Comment (slevenick) [reply mmv1/third_party/terraform/utils/common_diff_suppress.go.erb:185]: Yeah, it locks in whatever exists in state which allows the user to remove the field from their config without triggering recreate (as the entire resource is `input`/ForceNew)

**Pull request merged**